packages/terra-button/src/Button.jsx

```diff
@@ -6,105 +6,238 @@ import styles from './Button.scss';
    */
-  type: PropTypes.oneOf(['button', 'submit', 'reset']),
+  onKeyDown: PropTypes.func,
   /**
-   * Sets the button variant. One of `primary`, `secondary`, or `link`
+   * Callback function triggered when key is released.
    */
-  variant: PropTypes.oneOf(['default', 'link', 'primary', 'secondary']),
+  onKeyUp: PropTypes.func,
+  /**
+   * Sets the button text. If the button `isIconOnly` or of variant `ButtonVariants.UTILITY`, this text is set as the aria-label for accessiblity.
+   */
+  text: PropTypes.string.isRequired,
+  /**
+   * Sets the button type. One of `ButtonTypes.BUTTON`, `ButtonTypes.SUBMIT`, or `ButtonTypes.RESET`.
+   */
+  type: PropTypes.oneOf([ButtonTypes.BUTTON, ButtonTypes.SUBMIT, ButtonTypes.RESET]),
+  /**
+   * Sets the button variant. One of `ButtonVariants.NEUTRAL`,  `ButtonVariants.EMPHASIS`, `ButtonVariants['DE-EMPSHASIS']`, `ButtonVariants.ACTION` or `ButtonVariants.UTILITY`.
+   */
+  variant: PropTypes.oneOf([ButtonVariants.NEUTRAL, ButtonVariants.EMPHASIS, ButtonVariants['DE-EMPSHASIS'], ButtonVariants.ACTION, ButtonVariants.UTILITY]),
 };
 
 const defaultProps = {
   isBlock: false,
-  isCompact: false,
   isDisabled: false,
+  isIconOnly: false,
   isReversed: false,
-  type: 'button',
-  variant: 'default',
+  type: ButtonTypes.BUTTON,
+  variant: ButtonVariants.NEUTRAL,
 };
 
-const Button = ({
-  children,
-  icon,
-  isBlock,
-  isCompact,
-  isDisabled,
-  isReversed,
-  size,
-  text,
-  type,
-  variant,
-  ...customProps
-  }) => {
-  const attributes = Object.assign({}, customProps);
-  const buttonText = text ? <span className={styles.text}>{text}</span> : null;
-
-  attributes.className = cx([
-    'button',
-    variant,
-    { 'is-disabled': isDisabled },
-    size,
-    { block: isBlock },
-    { compact: isCompact },
-    attributes.className,
-  ]);
-
-  attributes.type = type;
-  attributes.disabled = isDisabled;
-  attributes.tabIndex = isDisabled ? '-1' : undefined;
-  attributes['aria-disabled'] = isDisabled;
-
-  const order = isReversed ?
-    [buttonText, icon, children] :
-    [icon, buttonText, children];
-
-  return React.createElement(attributes.href ? 'a' : 'button', attributes, ...order);
-};
+class Button extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { active: false, focused: false, mouseWasClicked: false };
+    this.handleKeyDown = this.handleKeyDown.bind(this);
+    this.handleKeyUp = this.handleKeyUp.bind(this);
+    this.handleOnBlur = this.handleOnBlur.bind(this);
+  }
+
+  handleOnBlur(event) {
+    this.setState({ focused: false });
+
+    if (this.props.onBlur) {
+      this.props.onBlur(event);
+    }
+  }
+
+  handleKeyDown(event) {
+    // Add active state to FF browsers
+    if (event.nativeEvent.keyCode === KEYCODES.SPACE) {
+      this.setState({ active: true });
+
+      // Follow href on space keydown when rendered as an anchor tag
+      if (this.props.href) {
+        // Prevent window scrolling
```

Comment:
What window scrolling is this preventing?

Reply:
This is preventing window scroll on space down.

By default, anchor tags will not redirect a user on space down, but I am manually triggering it so the anchor buttons ask like the other button types.